frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -54,6 +55,58 @@ import {
 import './ocs-storage-class-form.scss';
 import { GUARDED_FEATURES } from '../../features';
 
+export const CephFsNameComponent: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+
+  const [externalFsName, setExternalFsName] = React.useState('');
+
+  const [cephClusters, loaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+
+  const onExternalFsNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const name = e.currentTarget.value;
+    setExternalFsName(name);
+    onParamChange(name);
+  };
+
+  React.useEffect(() => {
+    if (!loaded || loadError) onParamChange('');
+    if (loaded && !loadError && cephClusters[0]?.metadata.name === CEPH_INTERNAL_CR_NAME) {
+      onParamChange(OCS_STORAGECLUSTER_CEPHFILESYSTEM);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [cephClusters, loaded, loadError]);
```

Comment:
```suggestion
const isExternal = useFlag(OCS_INDEPENDENT_FLAG);
const clusterName = !isExternal ? CEPH_INTERNAL_CR_NAME : CEPH_EXTERNAL_CR_NAME;
```

Reply:
I think `useEffect` is required here, we have to pass the default fs name once to `onParamChange` (for internal mode) otherwise `create` button will remain disabled.

Reply:
Makes sense we can extract that part and retain the `useEffect`